### ISSUE TITLE
Muliple errors compiling on Ubuntu 20.04

### ISSUE THREAD

**Issue by PerceptMD** [SMRT-AIST/fast_gicp#32]
## System
Ubuntu 20.04
PCL 1.11
gcc/g++ version 7.5.0 (Ubuntu 7.5.0-6ubuntu2) 
boost 1.71.0.0ubuntu2
## Steps done
```
git clone https://github.com/SMRT-AIST/fast_gicp --recursive
git submodule update --init --recursive
mkdir fast_gicp/build
cd fast_gicp/build
cmake .. -DCMAKE_BUILD_TYPE=Release
make # no -j to keep amount of printed errors small
```
## Errors
```
[  8%] Linking CXX shared library devel/lib/libfast_vgicp_cuda.so
[ 16%] Built target fast_vgicp_cuda
[ 25%] Building CXX object CMakeFiles/fast_gicp.dir/src/fast_gicp/gicp/fast_gicp.cpp.o
In file included from software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:1:0:
software/Software-local/fast_gicp/include/fast_gicp/gicp/fast_gicp.hpp:75:43: error: ‘shared_ptr’ in namespace ‘boost’ does not name a template type
   bool calculate_covariances(const boost::shared_ptr<const pcl::PointCloud<PointT>>& cloud, pcl::search::KdTree<PointT>& kdtree, std::vector<Matrix4, Eigen::aligned_allocator<Matrix4>>& covariances);
                                           ^~~~~~~~~~
software/Software-local/fast_gicp/include/fast_gicp/gicp/fast_gicp.hpp:75:53: error: expected ‘,’ or ‘...’ before ‘<’ token
   bool calculate_covariances(const boost::shared_ptr<const pcl::PointCloud<PointT>>& cloud, pcl::search::KdTree<PointT>& kdtree, std::vector<Matrix4, Eigen::aligned_allocator<Matrix4>>& covariances);
                                                     ^
In file included from software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:2:0:
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp: In member function ‘Eigen::VectorXf fast_gicp::FastGICP<PointSource, PointTarget>::loss_ls(const Eigen::Matrix<float, 6, 1>&, Eigen::MatrixXf*) const’:
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:201:8: error: ‘atomic_int’ is not a member of ‘std’
   std::atomic_int count(0);
        ^~~~~~~~~~
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:201:8: note: suggested alternative: ‘atomic_load’
   std::atomic_int count(0);
        ^~~~~~~~~~
        atomic_load
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:232:13: error: ‘count’ was not declared in this scope
     int n = count++;
             ^~~~~
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:232:13: note: suggested alternative:
In file included from /usr/include/c++/7/algorithm:62:0,
                 from /usr/include/eigen3/Eigen/Core:288,
                 from software/Software-local/fast_gicp/include/fast_gicp/gicp/fast_gicp.hpp:4,
                 from software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:1:
/usr/include/c++/7/bits/stl_algo.h:4076:5: note:   ‘std::count’
     count(_InputIterator __first, _InputIterator __last, const _Tp& __value)
     ^~~~~
In file included from software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:2:0:
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:238:20: error: ‘count’ was not declared in this scope
   int final_size = count;
                    ^~~~~
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:238:20: note: suggested alternative:
In file included from /usr/include/c++/7/algorithm:62:0,
                 from /usr/include/eigen3/Eigen/Core:288,
                 from software/Software-local/fast_gicp/include/fast_gicp/gicp/fast_gicp.hpp:4,
                 from software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:1:
/usr/include/c++/7/bits/stl_algo.h:4076:5: note:   ‘std::count’
     count(_InputIterator __first, _InputIterator __last, const _Tp& __value)
     ^~~~~
In file included from software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:2:0:
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp: At global scope:
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:245:77: error: ‘shared_ptr’ in namespace ‘boost’ does not name a template type
 tGICP<PointSource, PointTarget>::calculate_covariances(const boost::shared_ptr<const pcl::PointCloud<PointT>>& cloud, pcl::search::KdTree<PointT>& kdtree, std::vector<Matrix4, Eigen::aligned_allocator<Matrix4>>& covariances) {
                                                                     ^~~~~~~~~~
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:245:87: error: expected ‘,’ or ‘...’ before ‘<’ token
 tSource, PointTarget>::calculate_covariances(const boost::shared_ptr<const pcl::PointCloud<PointT>>& cloud, pcl::search::KdTree<PointT>& kdtree, std::vector<Matrix4, Eigen::aligned_allocator<Matrix4>>& covariances) {
                                                                     ^
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp: In member function ‘bool fast_gicp::FastGICP<PointSource, PointTarget>::calculate_covariances(int)’:
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:246:3: error: ‘kdtree’ was not declared in this scope
   kdtree.setInputCloud(cloud);
   ^~~~~~
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:246:3: note: suggested alternative: ‘free’
   kdtree.setInputCloud(cloud);
   ^~~~~~
   free
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:246:24: error: ‘cloud’ was not declared in this scope
   kdtree.setInputCloud(cloud);
                        ^~~~~
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:246:24: note: suggested alternative: ‘close’
   kdtree.setInputCloud(cloud);
                        ^~~~~
                        close
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:247:3: error: ‘covariances’ was not declared in this scope
   covariances.resize(cloud->size());
   ^~~~~~~~~~~
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:250:3: error: invalid controlling predicate
   for(int i = 0; i < cloud->size(); i++) {
   ^~~
In file included from software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:2:0:
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp: In instantiation of ‘void fast_gicp::FastGICP<PointSource, PointTarget>::setInputSource(const PointCloudSourceConstPtr&) [with PointSource = pcl::PointXYZ; PointTarget = pcl::PointXYZ; fast_gicp::FastGICP<PointSource, PointTarget>::PointCloudSourceConstPtr = std::shared_ptr<const pcl::PointCloud<pcl::PointXYZ> >]’:
software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:4:27:   required from here
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:95:24: error: no matching function for call to ‘fast_gicp::FastGICP<pcl::PointXYZ, pcl::PointXYZ>::calculate_covariances(const PointCloudSourceConstPtr&, pcl::search::KdTree<pcl::PointXYZ, pcl::KdTreeFLANN<pcl::PointXYZ, flann::L2_Simple<float> > >&, std::vector<Eigen::Matrix<float, 4, 4>, Eigen::aligned_allocator<Eigen::Matrix<float, 4, 4> > >&)’
   calculate_covariances(cloud, *source_kdtree, source_covs);
   ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:1:0:
software/Software-local/fast_gicp/include/fast_gicp/gicp/fast_gicp.hpp:75:8: note: candidate: template<class PointT> bool fast_gicp::FastGICP<PointSource, PointTarget>::calculate_covariances(int) [with PointT = PointT; PointSource = pcl::PointXYZ; PointTarget = pcl::PointXYZ]
   bool calculate_covariances(const boost::shared_ptr<const pcl::PointCloud<PointT>>& cloud, pcl::search::KdTree<PointT>& kdtree, std::vector<Matrix4, Eigen::aligned_allocator<Matrix4>>& covariances);
        ^~~~~~~~~~~~~~~~~~~~~
software/Software-local/fast_gicp/include/fast_gicp/gicp/fast_gicp.hpp:75:8: note:   template argument deduction/substitution failed:
In file included from software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:2:0:
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:95:24: note:   candidate expects 1 argument, 3 provided
   calculate_covariances(cloud, *source_kdtree, source_covs);
   ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp: In instantiation of ‘void fast_gicp::FastGICP<PointSource, PointTarget>::setInputTarget(const PointCloudTargetConstPtr&) [with PointSource = pcl::PointXYZ; PointTarget = pcl::PointXYZ; fast_gicp::FastGICP<PointSource, PointTarget>::PointCloudTargetConstPtr = std::shared_ptr<const pcl::PointCloud<pcl::PointXYZ> >]’:
software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:4:27:   required from here
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:104:24: error: no matching function for call to ‘fast_gicp::FastGICP<pcl::PointXYZ, pcl::PointXYZ>::calculate_covariances(const PointCloudTargetConstPtr&, pcl::search::KdTree<pcl::PointXYZ, pcl::KdTreeFLANN<pcl::PointXYZ, flann::L2_Simple<float> > >&, std::vector<Eigen::Matrix<float, 4, 4>, Eigen::aligned_allocator<Eigen::Matrix<float, 4, 4> > >&)’
   calculate_covariances(cloud, *target_kdtree, target_covs);
   ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:1:0:
software/Software-local/fast_gicp/include/fast_gicp/gicp/fast_gicp.hpp:75:8: note: candidate: template<class PointT> bool fast_gicp::FastGICP<PointSource, PointTarget>::calculate_covariances(int) [with PointT = PointT; PointSource = pcl::PointXYZ; PointTarget = pcl::PointXYZ]
   bool calculate_covariances(const boost::shared_ptr<const pcl::PointCloud<PointT>>& cloud, pcl::search::KdTree<PointT>& kdtree, std::vector<Matrix4, Eigen::aligned_allocator<Matrix4>>& covariances);
        ^~~~~~~~~~~~~~~~~~~~~
software/Software-local/fast_gicp/include/fast_gicp/gicp/fast_gicp.hpp:75:8: note:   template argument deduction/substitution failed:
In file included from software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:2:0:
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:104:24: note:   candidate expects 1 argument, 3 provided
   calculate_covariances(cloud, *target_kdtree, target_covs);
   ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp: In instantiation of ‘void fast_gicp::FastGICP<PointSource, PointTarget>::setInputSource(const PointCloudSourceConstPtr&) [with PointSource = pcl::PointXYZI; PointTarget = pcl::PointXYZI; fast_gicp::FastGICP<PointSource, PointTarget>::PointCloudSourceConstPtr = std::shared_ptr<const pcl::PointCloud<pcl::PointXYZI> >]’:
software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:5:27:   required from here
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:95:24: error: no matching function for call to ‘fast_gicp::FastGICP<pcl::PointXYZI, pcl::PointXYZI>::calculate_covariances(const PointCloudSourceConstPtr&, pcl::search::KdTree<pcl::PointXYZI, pcl::KdTreeFLANN<pcl::PointXYZI, flann::L2_Simple<float> > >&, std::vector<Eigen::Matrix<float, 4, 4>, Eigen::aligned_allocator<Eigen::Matrix<float, 4, 4> > >&)’
   calculate_covariances(cloud, *source_kdtree, source_covs);
   ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:1:0:
software/Software-local/fast_gicp/include/fast_gicp/gicp/fast_gicp.hpp:75:8: note: candidate: template<class PointT> bool fast_gicp::FastGICP<PointSource, PointTarget>::calculate_covariances(int) [with PointT = PointT; PointSource = pcl::PointXYZI; PointTarget = pcl::PointXYZI]
   bool calculate_covariances(const boost::shared_ptr<const pcl::PointCloud<PointT>>& cloud, pcl::search::KdTree<PointT>& kdtree, std::vector<Matrix4, Eigen::aligned_allocator<Matrix4>>& covariances);
        ^~~~~~~~~~~~~~~~~~~~~
software/Software-local/fast_gicp/include/fast_gicp/gicp/fast_gicp.hpp:75:8: note:   template argument deduction/substitution failed:
In file included from software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:2:0:
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:95:24: note:   candidate expects 1 argument, 3 provided
   calculate_covariances(cloud, *source_kdtree, source_covs);
   ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp: In instantiation of ‘void fast_gicp::FastGICP<PointSource, PointTarget>::setInputTarget(const PointCloudTargetConstPtr&) [with PointSource = pcl::PointXYZI; PointTarget = pcl::PointXYZI; fast_gicp::FastGICP<PointSource, PointTarget>::PointCloudTargetConstPtr = std::shared_ptr<const pcl::PointCloud<pcl::PointXYZI> >]’:
software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:5:27:   required from here
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:104:24: error: no matching function for call to ‘fast_gicp::FastGICP<pcl::PointXYZI, pcl::PointXYZI>::calculate_covariances(const PointCloudTargetConstPtr&, pcl::search::KdTree<pcl::PointXYZI, pcl::KdTreeFLANN<pcl::PointXYZI, flann::L2_Simple<float> > >&, std::vector<Eigen::Matrix<float, 4, 4>, Eigen::aligned_allocator<Eigen::Matrix<float, 4, 4> > >&)’
   calculate_covariances(cloud, *target_kdtree, target_covs);
   ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:1:0:
software/Software-local/fast_gicp/include/fast_gicp/gicp/fast_gicp.hpp:75:8: note: candidate: template<class PointT> bool fast_gicp::FastGICP<PointSource, PointTarget>::calculate_covariances(int) [with PointT = PointT; PointSource = pcl::PointXYZI; PointTarget = pcl::PointXYZI]
   bool calculate_covariances(const boost::shared_ptr<const pcl::PointCloud<PointT>>& cloud, pcl::search::KdTree<PointT>& kdtree, std::vector<Matrix4, Eigen::aligned_allocator<Matrix4>>& covariances);
        ^~~~~~~~~~~~~~~~~~~~~
software/Software-local/fast_gicp/include/fast_gicp/gicp/fast_gicp.hpp:75:8: note:   template argument deduction/substitution failed:
In file included from software/Software-local/fast_gicp/src/fast_gicp/gicp/fast_gicp.cpp:2:0:
software/Software-local/fast_gicp/include/fast_gicp/gicp/impl/fast_gicp_impl.hpp:104:24: note:   candidate expects 1 argument, 3 provided
   calculate_covariances(cloud, *target_kdtree, target_covs);
   ~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

**Comment by koide3** [comment]
It seems it has some problems with the latest PCL. I try to do some tests with the latest PCL to find the cause.

**Comment by PerceptMD** [comment]
Awesome!
I can provide you a PR fixing a few of these errors if you like to take a look.

**Comment by koide3** [comment]
Thanks for your offer. I fixed some errors related to the PCL version and pushed the updated code to ```refactor``` branch. Could you check if it works?

**Comment by PerceptMD** [comment]
```
cmake .. -DCMAKE_BUILD_TYPE=Release
make
```
Worked like a charm!


```
cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_VGICP_CUDA=ON
make
```
Sadly still fails:
```
<command-line>:0:20: warning: ISO C++11 requires whitespace after the macro name
Scanning dependencies of target fast_vgicp_cuda
[ 16%] Linking CXX shared library devel/lib/libfast_vgicp_cuda.so
[ 16%] Built target fast_vgicp_cuda
Scanning dependencies of target fast_gicp
[ 25%] Building CXX object CMakeFiles/fast_gicp.dir/src/fast_gicp/gicp/fast_gicp_st.cpp.o
[ 33%] Building CXX object CMakeFiles/fast_gicp.dir/src/fast_gicp/gicp/fast_gicp.cpp.o
[ 41%] Building CXX object CMakeFiles/fast_gicp.dir/src/fast_gicp/gicp/fast_vgicp.cpp.o
[ 50%] Building CXX object CMakeFiles/fast_gicp.dir/src/fast_gicp/gicp/experimental/fast_gicp_mp.cpp.o
[ 58%] Building CXX object CMakeFiles/fast_gicp.dir/src/fast_gicp/gicp/fast_vgicp_cuda.cpp.o
cc1plus: warning: /usr/lib/x86_64-linux-gnu/libz.so: not a directory
cc1plus: warning: /usr/lib/x86_64-linux-gnu/libz.so: not a directory
cc1plus: warning: /usr/lib/x86_64-linux-gnu/libz.so: not a directory
cc1plus: warning: /usr/lib/x86_64-linux-gnu/libz.so: not a directory
cc1plus: warning: /usr/lib/x86_64-linux-gnu/libz.so: not a directory
[ 66%] Linking CXX shared library devel/lib/libfast_gicp.so
[ 66%] Built target fast_gicp
Scanning dependencies of target gicp_align
Scanning dependencies of target gicp_kitti
[ 75%] Building CXX object CMakeFiles/gicp_align.dir/src/align.cpp.o
[ 83%] Building CXX object CMakeFiles/gicp_kitti.dir/src/kitti.cpp.o
cc1plus: warning: /usr/lib/x86_64-linux-gnu/libz.so: not a directory
cc1plus: warning: /usr/lib/x86_64-linux-gnu/libz.so: not a directory
..fast_gicp/src/kitti.cpp: In function ‘int main(int, char**)’:
..fast_gicp/src/kitti.cpp:95:24: error: ‘make_shared’ is not a member of ‘boost’
   auto target = boost::make_shared<pcl::PointCloud<pcl::PointXYZ>>();
                        ^~~~~~~~~~~
..fast_gicp/src/kitti.cpp:95:24: note: suggested alternatives:
In file included from /usr/include/c++/7/memory:81:0,
                 from /usr/include/boost/config/no_tr1/memory.hpp:21,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:23,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from /usr/include/boost/format/alt_sstream.hpp:21,
                 from /usr/include/boost/format/internals.hpp:24,
                 from /usr/include/boost/format.hpp:38,
                 from ..fast_gicp/src/kitti.cpp:3:
/usr/include/c++/7/bits/shared_ptr.h:703:5: note:   ‘std::make_shared’
     make_shared(_Args&&... __args)
     ^~~~~~~~~~~
In file included from /usr/local/include/pcl-1.11/pcl/impl/point_types.hpp:41:0,
                 from /usr/local/include/pcl-1.11/pcl/point_types.h:349,
                 from ..fast_gicp/src/kitti.cpp:7:
/usr/local/include/pcl-1.11/pcl/memory.h:122:66: note:   ‘pcl::make_shared’
 std::enable_if_t<!has_custom_allocator<T>::value, shared_ptr<T>> make_shared(Args&&... args)
                                                                  ^~~~~~~~~~~
..fast_gicp/src/kitti.cpp:95:65: error: expected primary-expression before ‘>’ token
   auto target = boost::make_shared<pcl::PointCloud<pcl::PointXYZ>>();
                                                                 ^~
..fast_gicp/src/kitti.cpp:95:68: error: expected primary-expression before ‘)’ token
   auto target = boost::make_shared<pcl::PointCloud<pcl::PointXYZ>>();
                                                                    ^
..fast_gicp/src/kitti.cpp:104:28: error: ‘make_shared’ is not a member of ‘boost’
   auto trajectory = boost::make_shared<pcl::PointCloud<pcl::PointXYZ>>();
                            ^~~~~~~~~~~
..fast_gicp/src/kitti.cpp:104:28: note: suggested alternatives:
In file included from /usr/include/c++/7/memory:81:0,
                 from /usr/include/boost/config/no_tr1/memory.hpp:21,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:23,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from /usr/include/boost/format/alt_sstream.hpp:21,
                 from /usr/include/boost/format/internals.hpp:24,
                 from /usr/include/boost/format.hpp:38,
                 from ..fast_gicp/src/kitti.cpp:3:
/usr/include/c++/7/bits/shared_ptr.h:703:5: note:   ‘std::make_shared’
     make_shared(_Args&&... __args)
     ^~~~~~~~~~~
In file included from /usr/local/include/pcl-1.11/pcl/impl/point_types.hpp:41:0,
                 from /usr/local/include/pcl-1.11/pcl/point_types.h:349,
                 from ..fast_gicp/src/kitti.cpp:7:
/usr/local/include/pcl-1.11/pcl/memory.h:122:66: note:   ‘pcl::make_shared’
 std::enable_if_t<!has_custom_allocator<T>::value, shared_ptr<T>> make_shared(Args&&... args)
                                                                  ^~~~~~~~~~~
..fast_gicp/src/kitti.cpp:104:69: error: expected primary-expression before ‘>’ token
   auto trajectory = boost::make_shared<pcl::PointCloud<pcl::PointXYZ>>();
                                                                     ^~
..fast_gicp/src/kitti.cpp:104:72: error: expected primary-expression before ‘)’ token
   auto trajectory = boost::make_shared<pcl::PointCloud<pcl::PointXYZ>>();
                                                                        ^
..fast_gicp/src/kitti.cpp:117:26: error: ‘make_shared’ is not a member of ‘boost’
     auto source = boost::make_shared<pcl::PointCloud<pcl::PointXYZ>>();
                          ^~~~~~~~~~~
..fast_gicp/src/kitti.cpp:117:26: note: suggested alternatives:
In file included from /usr/include/c++/7/memory:81:0,
                 from /usr/include/boost/config/no_tr1/memory.hpp:21,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:23,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from /usr/include/boost/format/alt_sstream.hpp:21,
                 from /usr/include/boost/format/internals.hpp:24,
                 from /usr/include/boost/format.hpp:38,
                 from ..fast_gicp/src/kitti.cpp:3:
/usr/include/c++/7/bits/shared_ptr.h:703:5: note:   ‘std::make_shared’
     make_shared(_Args&&... __args)
     ^~~~~~~~~~~
In file included from /usr/local/include/pcl-1.11/pcl/impl/point_types.hpp:41:0,
                 from /usr/local/include/pcl-1.11/pcl/point_types.h:349,
                 from ..fast_gicp/src/kitti.cpp:7:
/usr/local/include/pcl-1.11/pcl/memory.h:122:66: note:   ‘pcl::make_shared’
 std::enable_if_t<!has_custom_allocator<T>::value, shared_ptr<T>> make_shared(Args&&... args)
                                                                  ^~~~~~~~~~~
..fast_gicp/src/kitti.cpp:117:67: error: expected primary-expression before ‘>’ token
     auto source = boost::make_shared<pcl::PointCloud<pcl::PointXYZ>>();
                                                                   ^~
..fast_gicp/src/kitti.cpp:117:70: error: expected primary-expression before ‘)’ token
     auto source = boost::make_shared<pcl::PointCloud<pcl::PointXYZ>>();
                                                                      ^
..fast_gicp/src/kitti.cpp:122:27: error: ‘make_shared’ is not a member of ‘boost’
     auto aligned = boost::make_shared<pcl::PointCloud<pcl::PointXYZ>>();
                           ^~~~~~~~~~~
..fast_gicp/src/kitti.cpp:122:27: note: suggested alternatives:
In file included from /usr/include/c++/7/memory:81:0,
                 from /usr/include/boost/config/no_tr1/memory.hpp:21,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:23,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from /usr/include/boost/format/alt_sstream.hpp:21,
                 from /usr/include/boost/format/internals.hpp:24,
                 from /usr/include/boost/format.hpp:38,
                 from ..fast_gicp/src/kitti.cpp:3:
/usr/include/c++/7/bits/shared_ptr.h:703:5: note:   ‘std::make_shared’
     make_shared(_Args&&... __args)
     ^~~~~~~~~~~
In file included from /usr/local/include/pcl-1.11/pcl/impl/point_types.hpp:41:0,
                 from /usr/local/include/pcl-1.11/pcl/point_types.h:349,
                 from ..fast_gicp/src/kitti.cpp:7:
/usr/local/include/pcl-1.11/pcl/memory.h:122:66: note:   ‘pcl::make_shared’
 std::enable_if_t<!has_custom_allocator<T>::value, shared_ptr<T>> make_shared(Args&&... args)
                                                                  ^~~~~~~~~~~
..fast_gicp/src/kitti.cpp:122:68: error: expected primary-expression before ‘>’ token
     auto aligned = boost::make_shared<pcl::PointCloud<pcl::PointXYZ>>();
                                                                    ^~
..fast_gicp/src/kitti.cpp:122:71: error: expected primary-expression before ‘)’ token
     auto aligned = boost::make_shared<pcl::PointCloud<pcl::PointXYZ>>();
                                                                       ^
make[2]: *** [CMakeFiles/gicp_kitti.dir/build.make:82: CMakeFiles/gicp_kitti.dir/src/kitti.cpp.o] Fehler 1
make[1]: *** [CMakeFiles/Makefile2:413: CMakeFiles/gicp_kitti.dir/all] Fehler 2
make[1]: *** Auf noch nicht beendete Prozesse wird gewartet …
[ 91%] Linking CXX executable devel/lib/fast_gicp/gicp_align
[ 91%] Built target gicp_align
make: *** [Makefile:160: all] Fehler 2
```
